include/zephyr/kernel.h

```diff
@@ -6153,6 +6153,186 @@ void k_sys_runtime_stats_enable(void);
  */
 void k_sys_runtime_stats_disable(void);
 
+/**
+ * @defgroup k_sig Kernel Signal APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/**
+ * @brief Process ID
+ *
+ * @note Process IDs are notional at this time, as processes are unsupported in Zephyr.
+ */
+typedef k_tid_t k_pid_t;
+
+/** @brief Signal Values */
+union k_sig_val {
+	int sival_int;
+	void *sival_ptr;
+};
+
+/** @brief Signal info */
+struct k_sig_info {
+	int si_signo;
+	int si_code;
+	union k_sig_val si_value;
+};
+
+/** @brief A bitset large enough to contain all signal bits. */
+struct k_sig_set;
+
+#define K_SIG_BLOCK   0
+#define K_SIG_SETMASK 1
+#define K_SIG_UNBLOCK 2
+
+/**
+ * @brief Add a signal @a sig to signal set @a set.
```

Comment:
I have no hard preference, and could definitely switch to `k_sig` instead of `k_sig_`. Whatever is easier to remember.

Definitely also happy to improve docs.

Reply:
If we want to namespace this API, then it might be better to stick with `k_sig_xxx`